app/components/tables/holidays_table/HolidaysTable.tsx

```diff
@@ -0,0 +1,130 @@
+import {
+    useReactTable,
+    flexRender,
+    getCoreRowModel,
+    getPaginationRowModel,
+    getFilteredRowModel,
+    getSortedRowModel,
+} from "@tanstack/react-table";
+import useAxiosPromise from "../../../hooks/useAxiosPromise";
+import React, { useMemo, useState, useEffect } from "react";
+import HolidaysTableHeaders from "@/components/tables/holidays_table/table_element/HolidaysTableHeaders";
+import { Holiday } from "./Holiday";
+import SortArrow from "@/components/tables/holidays_table/table_element/SortArrow";
+import PaginationBar from "@/components/tables/holidays_table/table_element/PaginationBar";
+import FilterBar from "@/components/tables/holidays_table/table_element/FilterBar";
+interface HolidaysTableProps {
+    data: Holiday[];
+}
+
+
+const HolidaysTable = ({ data }: HolidaysTableProps) => {
+    const [filtering, setFiltering] = useState<string>("");
+    const [firstColumnSorting, setFirstColumnSorting] = useState<'asc' | 'desc'>('asc');
+    const [sorting, setSorting] = useState([{ id: 'description', desc: firstColumnSorting === 'desc' }]);
+    const columns = HolidaysTableHeaders;
+
+    const tableInstance = useReactTable({
+        columns,
```

Comment:
We can say `columns: HolidaysTableHeader` instead of declaring a `columns` variable.